alert-database/src/test/java/com/synopsys/integration/alert/database/api/DefaultDescriptorAccessorTest.java

```diff
@@ -0,0 +1,289 @@
+
+        Mockito.when(descriptorTypeRepository.findFirstByType(descriptorType.name())).thenReturn(Optional.empty());
+        Mockito.when(descriptorTypeRepository.save(Mockito.any())).thenReturn(descriptorTypeEntity);
+        Mockito.when(registeredDescriptorRepository.findByTypeId(Mockito.any())).thenReturn(List.of(registeredDescriptorEntity));
+        Mockito.when(descriptorTypeRepository.findById(registeredDescriptorEntity.getTypeId())).thenReturn(Optional.of(descriptorTypeEntity));
+
+        DefaultDescriptorAccessor descriptorAccessor = new DefaultDescriptorAccessor(registeredDescriptorRepository, null, null, descriptorTypeRepository);
+        List<RegisteredDescriptorModel> registeredDescriptorModelList = descriptorAccessor.getRegisteredDescriptorsByType(descriptorType);
+
+        assertEquals(1, registeredDescriptorModelList.size());
+        assertEquals(name, registeredDescriptorModelList.get(0).getName());
+        assertEquals(descriptorType, registeredDescriptorModelList.get(0).getType());
+    }
+
+    @Test
+    public void descriptorTypeFailureTest() throws Exception {
+        DefaultDescriptorAccessor descriptorAccessor = new DefaultDescriptorAccessor(null, null, null, null);
+        DescriptorType descriptorType = null;
+
+        try {
+            descriptorAccessor.getRegisteredDescriptorsByType(descriptorType);
+            fail();
+        } catch (AlertDatabaseConstraintException e) {
+            logger.error(e.getMessage(), e);
+        }
+    }
+
+    @Test
+    public void getRegisteredDescriptorByIdTest() throws Exception {
+        final String name = "name-test";
+        final Long typeId = 1L;
+        final DescriptorType descriptorType = DescriptorType.CHANNEL;
+        final Long descriptorId = 2L;
+
+        RegisteredDescriptorEntity registeredDescriptorEntity = new RegisteredDescriptorEntity(name, typeId);
+        registeredDescriptorEntity.setId(1L);
+        DescriptorTypeEntity descriptorTypeEntity = new DescriptorTypeEntity(descriptorType.name());
+        descriptorTypeEntity.setId(2L);
+
+        RegisteredDescriptorRepository registeredDescriptorRepository = Mockito.mock(RegisteredDescriptorRepository.class);
+        DescriptorTypeRepository descriptorTypeRepository = Mockito.mock(DescriptorTypeRepository.class);
+
+        Mockito.when(registeredDescriptorRepository.findById(descriptorId)).thenReturn(Optional.of(registeredDescriptorEntity));
+        Mockito.when(descriptorTypeRepository.findById(Mockito.any())).thenReturn(Optional.of(descriptorTypeEntity));
+
+        DefaultDescriptorAccessor descriptorAccessor = new DefaultDescriptorAccessor(registeredDescriptorRepository, null, null, descriptorTypeRepository);
+        Optional<RegisteredDescriptorModel> registeredDescriptorModel = descriptorAccessor.getRegisteredDescriptorById(descriptorId);
+
+        assertTrue(registeredDescriptorModel.isPresent());
+        assertEquals(typeId, registeredDescriptorModel.get().getId());
```

Comment:
Instead of calling `registeredDescriptorModel.get()` over and over, you could create a local variable for the return of this method and use that variable instead. 
``` java
Optional<RegisteredDescriptorModel> registeredDescriptorModelOptional = descriptorAccessor.getRegisteredDescriptorById(descriptorId);
RegisteredDescriptorModel registeredDescriptorModel = registeredDescriptorModelOptional.get();
assertTrue(registeredDescriptorModel.isPresent());
assertEquals(typeId, registeredDescriptorModel.getId());
assertEquals(name, registeredDescriptorModel.getName());
assertEquals(descriptorType, registeredDescriptorModel.getType());
```